i6_models/util/mask.py

```diff
@@ -0,0 +1,16 @@
+import torch
+
+
+def tensor_mask_from_length(tensor: torch.Tensor, seq_len: torch.Tensor) -> torch.Tensor:
```

Comment:
The function does not really depend on `tensor` at all, except for `tensor.shape[1]`. I would only use that as an explicit arg `max_len`.

The corresponding function in TensorFlow is [`tf.sequence_mask`](https://www.tensorflow.org/api_docs/python/tf/sequence_mask), which has a much better name and API. Why don't we just follow that (more or less)? Specifically, I would suggest:

```suggestion
def sequence_mask(lens: torch.Tensor, /, *, max_len: Optional[torch.Tensor] = None) -> torch.Tensor:
```

The `max_len` can be optional. We can just use `torch.max(lens)` if it is not explicitly given.

Reply:
I had `torch.max(lens)` in the original function, and this was not traceable, so I proposed to disable masks for the export. Then you made the suggestion to use tensor.shape, because that is traceable. Why do you want to go back now?

Reply:
I did not suggest to change any logic here. I just say that `tensor` should not be the argument of the function, but `maxlen` (thus `tensor.shape[1]`) instead.

Also, I suggested to extend the logic by allowing `max_len=None` and having this default fallback logic.

And `torch.max` is traceable. It works just fine. Maybe you used `max` instead? This probably casts the `torch.Tensor` to a Python `int` and thus makes it non-traceable.

Reply:
Sorry for not being specific enough, I said using `torch.max` makes the whole function non-traceable, not that `torch.max` itself is not traceable. `torch.arange` only accepts an integer., so you would have to cast the result of torch.max to int, making it non-traceable. For some reason when using `tensor.shape` tracing works, although it is usually  treated like an integer, not a tensor.

Reply:
Hmm right, I wonder why the old tests failed. Yes, then the API change makes sense.

Reply:
What should I do for the device? Should we run the mask creation then on CPU and copy later, or pass device as extra argument?

Reply:
Ah good question. `device` should be an extra argument. But we probably want to allow for some default behavior. Using the device of `seq_len` is maybe not a good idea, as in practice, `seq_len` often will be on CPU, but we want the seq mask usually on GPU.

In Torch >=2, it might make sense to rely on the Torch default device behavior (using `with torch.device(...)` context scope or `torch.set_default_device(...)`), which you get automatically when you pass `device=None` (or not at all) to `torch.arange`. So maybe we can just leave it as that, and users of outdated PyTorch versions must explicitly pass `device`.